src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/SocketsHttpHandler.cs

```diff
@@ -345,7 +346,28 @@ private HttpMessageHandler SetupHandlerChain()
                 return Task.FromException<HttpResponseMessage>(error);
             }
 
-            return handler.SendAsync(request, cancellationToken);
+            return HttpTelemetry.IsEnabled && request.RequestUri != null ?
+                WithLogging(handler, request, cancellationToken) :
+                handler.SendAsync(request, cancellationToken);
+
+            static async Task<HttpResponseMessage> WithLogging(HttpMessageHandler handler, HttpRequestMessage request, CancellationToken cancellationToken)
+            {
+                Debug.Assert(request.RequestUri != null);
+                HttpTelemetry.Log.RequestStart(request.RequestUri.IdnHost, request.RequestUri.Port);
```

Comment:
Are these really the interesting fields? Yes Host is interesting, but I'd expect Scheme, Path, and protocol version to be more interesting than Port.

Reply:
How reasonable would it be to just log the full URI?

Reply:
EventSource events need simple fields. We can't use a complex type like System.Uri.

DiagnosticSource (like the DiagnosticHandler we have) can handle complex objects. But this telemetry is about EventSource/EventCounter.

Reply:
Uri has an AbsoluteUri string field that could be logged, or we can log some of the more interesting components like Path.

Reply:
That's fine. I was just responding to @scalablecory comment about the "full URI" since EventSource methods (events) need to have simpler arguments since they get emited that way in the traces.

Reply:
I pushed a commit to add more fields to the 'RequestStart' event. I also removed the fields from the RequestStop since RequestStart/RequestStop form an activity pair and the data can be inferred. This also helps makes the events more concise for perf considerations.